index.js

```diff
@@ -157,6 +151,14 @@ function defaultMethodNamesOrAssert(methodNames) {
 
 const isMetaEvent = eventName => eventName === listenerAdded || eventName === listenerRemoved;
 
+function emitMetaEvent(emitter, eventName, eventData) {
+	if (isMetaEvent(eventName)) {
+		canEmitMetaEvents = true;
+		emitter.emit(eventName, eventData);
```

Comment:
What should we do if this throws an error? `canEmitMetaEvents` won't be reverted to false...

Reply:
Use try/finally? And set it to `false` in the `finally` block.